che-theia-task-extension/src/node/messaging/connection.ts

```diff
@@ -0,0 +1,42 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2018 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { MessageConnection, Logger } from 'vscode-jsonrpc';
+import { createWebSocketConnection, IWebSocket } from 'vscode-ws-jsonrpc';
+import { ConsoleLogger } from 'vscode-ws-jsonrpc';
+import * as ReconnectingWebSocket from 'reconnecting-websocket';
+
+// copied from the https://github.com/TypeFox/vscode-ws-jsonrpc/blob/v0.0.2-2/src/connection.ts
```

Comment:
we can't re-use this file from theia without copying it ?

Reply:
We can't because of the original functions rely on the browser's WebSocket. And I need it for node env.

Reply:
is there an alternate library that we could use ?

Reply:
I've not tried to find. Just saw that piece of code in Theia that almost satisfies my needs.